common/lib/capa/capa/responsetypes.py

```diff
@@ -1523,6 +1757,41 @@ def get_score(self, student_answers):
         except Exception:
             raise general_exception
         # End `evaluator` block -- we figured out the student's answer!
+
+        tree = self.xml
+        problem_xml = tree.xpath('.')
+
+        # Partial credit type - can set 'close' or 'list'
+        credit_type = problem_xml[0].get('partial_credit', default=False)
+
+        # Allowing for multiple partial credit types. Divide on commas, strip whitespace.
+        if credit_type:
+            credit_type = credit_type.split(',')
+            credit_type = [word.strip().lower() for word in credit_type]
+
+        # What multiple of the tolerance is worth partial credit?
+        has_partial_range = tree.xpath('responseparam[@partial-range]')
+        if has_partial_range:
+            partial_range = has_partial_range[0].get('partial-range', default='2')
+            # Keep only digits in case people want to write 'x2' or '2x'
+            partial_range = float(re.sub(r'\D', '', partial_range))
```

Comment:
That's dangerous behavior for a number of reasons: 

1. If we have garbage, it will accept it. 2hkdhkljwhlk3lkjl should not be okay. Neither should "2 percent" 
2. It is inflexible. If we decide to add "2 percent" later, or 2M, or 2dB, or otherwise, we may end up with backwards compatibility issues. 

It's fine to check for 'x' specifically, but not the generic regexp. 

Nit: regexp should be compiled once, not in-line. If you use regexp, re.compile as a class or file level variable.